sensor/kubernetes/certrefresh/localscanner/certificate_requester.go

```diff
@@ -70,7 +76,7 @@ func (r *certificateRequesterImpl) dispatchResponses() {
 			}
 			r.requests.Delete(msg.GetRequestId())
 			// Doesn't block even if the corresponding call to RequestCertificates is cancelled and no one
-			// ever reads this, because requestC has buffer of 1, and we removed it from `r.request` above,
+			// ever reads this, because requestC has buffer of 1, and we removed it from `r.requests` above,
 			// in case we get more than 1 response for `msg.GetRequestId()`.
 			responseC.(chan *central.IssueLocalScannerCertsResponse) <- msg
```

Comment:
Oh this will block or the message will be dropped when Sensor is offline. We should add tests for this component that include operation in the offline mode and transitioning between modes (something happens in online, then we go offline for some time, then go back online and the logic should still work).

Reply:
I agree that this can be improved, but I actually don't think this is blocking, but might be confusing because of the naming.

We have several channels here:
* `r.receiveC` -> messages received from Central
* `r.sendC` -> messages to be sent to Central
* `responseC` is used internally by the requester, to pass messages that were received on `r.receiveC` in a goroutine. It's a buffered message of size 1 and we make sure to only write once in it. Doesn't help that at creation time this channel is called `receiveC` in the same file.

It also won't be dropped if Sensor is offline. If we got to this point, we already have the message from Central. It's just passing the message from one goroutine to another.